File: components/gatling-context/src/main/java/io/microsamples/gatlingrunner/load/GatlingContext.java
package io.microsamples.gatlingrunner.load;

public enum GatlingContext {
    INSTANCE;

    public String payload;
    public String baseUrl;
    public String endpoint;
    public HttpMethod httpMethod;

    public int constantUsersPerSecond;
    public int constantUsersPerSecondDuration;

    public int rampUsersPerSecondMinimum;
    public int rampUsersPerSecondMaximum;
    public int rampUsersPerSecondDuration;
}

File: components/gatling-context/src/main/kotlin/io/microsamples/gatlingrunner/load/GatlingContext.kt
package io.microsamples.gatlingrunner.load

enum class GatlingContext {
    INSTANCE;

    var baseUrl: String? = null
    var endpoint: String? = null
    var httpMethod: HttpMethod? = null
    var payload: String? = null

    var constantUsersPerSecond = 0
    var constantUsersPerSecondDuration = 0
    var rampUsersPerSecondMinimum = 0
    var rampUsersPerSecondMaximum = 0
    var rampUsersPerSecondDuration = 0
}
